src/app/core/services/map-zoom.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { environment } from '../../../environments/environment';
+
+const MAX_ZOOM = 1;
+const MIN_ZOOM = 0.25;
+const DEFAULT_ZOOM_STEP = 0.25;
+const DEFAULT_ZOOM_FACTOR = environment.defaultZoomFactor;
+const ONE_HUNGRED = 100;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MapZoomService {
+  public readonly zoomFactor$ = new BehaviorSubject<number>(
+    DEFAULT_ZOOM_FACTOR
+  );
+
+  public readonly isMaxZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MAX_ZOOM
+  );
+
+  public readonly isMinZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MIN_ZOOM
+  );
+
+  public readonly isOneHungredPercentZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR * ONE_HUNGRED === ONE_HUNGRED
+  );
+
+  public readonly scale$ = this.zoomFactor$.pipe(
+    map((value) => Math.round(value * ONE_HUNGRED))
+  );
+
+  public readonly scaleTransform$ = this.zoomFactor$.pipe(
+    map((value) => `scale(${value})`)
+  );
+
+  public zoomIn(step: number = DEFAULT_ZOOM_STEP): void {
+    const value = this._zoomFactor;
```

Comment:
Please check, do we really need this constant `const value = this._zoomFactor;` ?